src/components/NewDraftWizard/WizardForms/WizardJSONSchemaParser.js

```diff
@@ -589,22 +589,40 @@ const FormTextField = ({
   // Case: DOI form - Affilation fields to be prefilled
   const prefilledFields = ["affiliationIdentifier", "schemeUri", "affiliationIdentifierScheme"]
   let watchAutocompleteFieldName = ""
-  let watchFieldValue = null
+  let prefilledValue = null
+
+  let fullNameValue = "" // Case: DOI form - Creators and Contributors' FullName
+
+  let disabled = false // boolean if inputValue is disabled
+
   if (openedDoiForm) {
     watchAutocompleteFieldName =
-      name.includes("affiliation") && prefilledFields.includes(lastPathItem)
-        ? path.slice(0, -1).join(".").concat(".", "name")
-        : ""
+      name.includes("affiliation") && prefilledFields.includes(lastPathItem) ? getPathName(path, "name") : ""
+
     // useWatch to watch any changes in form's fields
     const watchValues = useWatch()
+
     // check changes of value of autocompleteField from watchValues
-    watchFieldValue = watchAutocompleteFieldName ? get(watchValues, watchAutocompleteFieldName) : null
-  }
+    prefilledValue = watchAutocompleteFieldName ? get(watchValues, watchAutocompleteFieldName) : null
+
+    // If it's <creators>'s and <contributors>'s FullName field, watch the values of GivenName and LastName
+    const isFullNameField = (path[0] === "creators" || path[0] === "contributors") && path[2] === "name"
+
+    if (isFullNameField) {
+      const givenName = getPathName(path, "givenName")
+      const givenNameValue = get(watchValues, givenName) || ""
+      const lastName = getPathName(path, "familyName")
```

Comment:
maybe use `familyName` everywhere instead of `lastName` for consistency, as we did the same with `givenName` ?

Reply:
all `lastName` are changed to `familyName`.